### DISCUSSION TITLE
Question of End-to-end Learning with Autoencoder

### DISCUSSION THREAD

**Discussion by Mio-TW** [NVlabs/sionna#271] [Q&A] [answered]
I have the question in [https://nvlabs.github.io/sionna/examples/Autoencoder.html#Trainable-End-to-end-System:-RL-based-Training](url)

For the RL-based Training, why "-" is added in the tx_loss? It may cause the loss become negative value. 
If there is no negative sign, increasing the power of "p" will result in a larger BCE and consequently a larger tx_loss. However, a negative sign is added, the outcome become opposite. I am curious about the reason behind this.

Thank you ao much.

**Comment by faycalaa** [comment, marked as answer]
Hi,

The code you are referring to implements the algorithm from [this paper](https://arxiv.org/pdf/1812.05929.pdf) and more specifically equation (8) in the case where `\pi` corresponds to the density of a normal distribution. Taking the log of this density results in the implemented equation, where the "-" corresponds to the one that is in the exp(.) of the density.
